_posts/2023-04-13-denoising-diffusion/denoising-diffusion.Rmd

```diff
@@ -0,0 +1,208 @@
+
+[^1]: From @dieleman2022diffusion, an excellent high-level introduction
+    to diffusion models.
+
+As a final example, what if we aren't interested in synthesis, but would
+like to assess if a given piece of data could likely be part of some
+distribution? If so, flows might be an option.[^2]
+
+[^2]: For more information on this lesser-used family of algorithms, see
+    <https://blog.evjang.com/2018/01/nf1.html>.
+
+## Zooming in: Diffusion models
+
+Just like about every deep-learning architecture, diffusion models
+constitute a heterogeneous family. Here, let us just name a few of the
+most en-vogue members. [^3]
+
+[^3]: En vogue as of this writing (the usual caveat that comes with
+    using epitheta such as this).
+
+When, above, we said that the idea of diffusion models was to
+sequentially transform an input into noise, then sequentially de-noise
+it again, we left open how that transformation is operationalized. This,
+in fact, is one area where rivaling approaches tend to differ.
+@abs-2011-13456, for example, make use of a a stochastic differential
+equation (SDE) that maintains the desired distribution during the
+information-destroying forward phase. In stark contrast, other
+approaches, inspired by @ho2020, rely on Markov chains to realize state
+transitions. The variant introduced here -- @song2020 -- keeps the same
+spirit, but improves on efficiency.
+
+## Our implementation -- overview
+
+The [README](https://github.com/dfalbel/denoising-diffusion) provides a
+very thorough introduction, covering (almost) everything from
+theoretical background via implementation details to training procedure
+and tuning. Here, we just outline a few basic facts.
+
+As already hinted at above, all the work happens during the forward
+stage. The network takes two inputs, the images as well as information
+about the signal-to-noise ratio to be applied at every step in the
+corruption process. That information may be encoded in various ways,
+and is then embedded, in some form, into a higher-dimensional space more
+conducive to learning. Here is how that could look, for two different types of scheduling/embedding:
+
+```{r, echo=FALSE, eval=TRUE, fig.alt = "One below the other, two sequences where the original flower image gets transformed into noise at differing speed.", out.width="704px", out.height="128px"}
+knitr::include_graphics("images/forward.png")
+```
+
+Architecture-wise, inputs as well as intended outputs being images, the
+main workhorse is a U-Net. It forms part of a top-level model that, for
+each input image, creates corrupted versions, corresponding to the noise
+rates requested, and runs the U-Net on them. From what is returned, it
+tries to deduce the noise level that was governing each instance.
+Training then consists in getting those estimates to improve.
+
+
+Model trained, the reverse process -- image generation -- is
+straightforward: It consists in recursive de-noising according to the
```

Comment:
Again, here (and in a couple other spots), you've got hyphenated de-noising—I'm in favor of changing it in the title to match.